### DISCUSSION TITLE
Using BlockDisplays

### DISCUSSION THREAD

**Discussion by lightyuwu** [Minestom/Minestom#2412] [Help] [answered]
I've tried for 3 hours, but whatever I do, i can't get Block Display entites to display anything, as i have no idea how Minestom NBT works..
All I need is it to display any block.
I know that the NBT is like this: {block_state:{Name:"someBlockHere"}} but I don't know how to put that into code...

**Comment by mworzala** [comment, marked as answer]
```java
var entity = new Entity(EntityType.BLOCK_DISPLAY);
var meta = ((BlockDisplayMeta) entity.getEntityMeta());
meta.setBlockState(Block.STONE);
entity.setInstance(myInstance, myPosition);
```

**Reply by lightyuwu** [reply]
I never knew BlockDisplayMeta existed... The docs still do seem half-finished, but i'll try that

**Reply by xZeGoe** [reply]
(don't set meta's instance, but entity one)

**Reply by DasBabyPixel** [reply]
[Docs seem fine](https://minestom.net/docs/feature/entities#entity-meta) to me

**Reply by lightyuwu** [reply]
> (don't set meta's instance, but entity one)

Yeah, would've figured *that* out, at least after the first error

**Reply by lightyuwu** [reply]
> [Docs seem fine](https://minestom.net/docs/feature/entities#entity-meta) to me

It only Talks about Horse and Boat Meta,  so i thought they had the most special data, and the Others just default. Would probably be smart to have a Page that lists all of them
